translate/storage/yaml.py

```diff
@@ -218,17 +220,56 @@ def get_root_node(self):
         return result
 
     def _parse_dict(self, data, prev):
-        # Does this look like a plural?
-        if data and all(x in cldr_plural_categories for x in data):
-            # Ensure we have correct plurals ordering.
-            values = [data[item] for item in cldr_plural_categories if item in data]
+        """Ensure plurals are stored as a single unit and handle partial blanks properly."""
+        if isinstance(data, dict):
+            plural_keys = [k for k in data if k in cldr_plural_categories]
+
+            # ✅ If all plural categories are present but some are empty, we still need the unit
+            if plural_keys and len(plural_keys) == len(data):
+                if any(v is not None for v in data.values()):
+                    # At least one non-empty value, so create a unit
+                    yield (
+                        prev,
+                        {k: data[k] for k in cldr_plural_categories if k in data},
```

Comment:
This makes parsing return `dict` instead of `multistring`, that will break any Python API user because nobody expects `dict` here. This part was not covered by the tests, I've just added that in https://github.com/translate/translate/pull/5572.